source/compiler-core/slang-source-loc.cpp

```diff
@@ -573,15 +573,38 @@ int SourceFile::calcColumnIndex(int lineIndex, int offset, int tabSize)
 
 void SourceFile::setContents(ISlangBlob* blob)
 {
-    const UInt contentSize = blob->getBufferSize();
+    const UInt rawContentSize = blob->getBufferSize();
 
-    SLANG_ASSERT(contentSize == m_contentSize);
+    SLANG_ASSERT(rawContentSize == m_contentSize);
 
-    char const* contentBegin = (char const*)blob->getBufferPointer();
-    char const* contentEnd = contentBegin + contentSize;
+    Byte* rawContentBegin = (Byte*)blob->getBufferPointer();
 
-    m_contentBlob = blob;
-    m_content = UnownedStringSlice(contentBegin, contentEnd);
+    // Query the encoding type and discard the Unicode Byte-Order-Marker before decoding
+    size_t offset;
+    auto type = CharEncoding::determineEncoding(
+        rawContentBegin,
+        rawContentSize,
+        offset);
+    SLANG_ASSERT(rawContentSize >= offset);
+
+    List<char> decodedBuffer;
+    CharEncoding::getEncoding(type)->decode(
+        rawContentBegin + offset,
+        rawContentSize - offset,
+        decodedBuffer);
+    List<uint8_t> decodedBuffer2;
+    for(auto i : decodedBuffer)
+    {
+        decodedBuffer2.add(static_cast<uint8_t>(i));
+    }
+    m_contentBlob = ListBlob::moveCreate(decodedBuffer2);
```

Comment:
I think this should be `StringBlob` from `String`
```
    String decodedBuffer2;
    decodedBuffer2.prepareForAppend(decodedBuffer.getCount());
    for(auto i : decodedBuffer)
    {
        decodedBuffer2.append(i);
    }

    m_contentBlob = StringBlob::create(decodedBuffer2);
```

Reply:
Yes, that seems to have resolved the problems I was seeing. Thanks!